lib/commands/backup.js

```diff
@@ -0,0 +1,230 @@
+const Promise = require('bluebird');
+const createDebug = require('debug');
+const Zip = require('adm-zip');
+const walker = require('klaw-sync');
+const fs = require('fs-extra');
+const path = require('path');
+const errors = require('../').errors;
+
+const debug = createDebug('ghost-cli:backup');
+
+function absolutePath(inputPath) {
+    return path.isAbsolute(inputPath) ? inputPath : path.join(process.cwd(), inputPath);
+}
+
+class BackupCommand extends Command {
+    run(argv) {
+        const instance = this.system.getInstance();
+
+        return this.ui.listr([{
+            task: this.initialize,
+            title: 'Checking configuration'
+        }, {
+            title: 'Backing up extensions',
+            task: this.backupExtensions
+        }, {
+            title: 'Backing up database',
+            task: this.backupDatabase
+        }, {
+            title: 'Backing up config files',
+            task: this.backupConfig
+        }, {
+            title: 'Backing up content folder',
+            task: this.backupContent.bind(this)
+        }, {
+            title: 'Writing backup file',
+            task: this.writeZipFile
+        }],{
+            system: this.system,
+            argv: argv,
+            instance: instance
+        });
+    }
+
+    initialize(ctx) {
+        const datetime = (new Date()).toJSON().substring(0, 10);
+
+        ctx.database = {};
+        ctx.saveLocation = process.cwd();
+
+        if (ctx.argv.output) {
+            ctx.saveLocation = absolutePath(ctx.argv.output);
+        }
+
+        try {
+            fs.ensureDirSync(ctx.saveLocation);
+            // After we're sure the directory exists, check to make sure we can write to it
+            fs.accessSync(ctx.saveLocation, fs.W_OK);
+        } catch (error) {
+            ctx.instance.ui.log('Ghost doesn\'t have permission to write to the output folder', 'red');
+            return Promise.reject(new errors.SystemError(error));
+        }
+
+        const saveRoot = ctx.saveLocation
+        ctx.saveLocation = path.join(saveRoot, `${ctx.instance.name}.backup.${datetime}.zip`);
+
+        let offset = 1;
+
+        while (fs.existsSync(ctx.saveLocation)) {
+            ctx.saveLocation = path.join(saveRoot, `${ctx.instance.name}.backup.${datetime}-${offset}.zip`);
+            offset += 1;
+        }
+
+        // The zip library used (adm-zip) has a known issue where the zip is structurally sound,
+        // but Windows has issues reading / unzipping it. To counteract this, all files are read
+        // and added via the `addFile` method, where the file permissions can be set, seemingly
+        // resolving the issue. See https://github.com/cthackers/adm-zip/issues/171
+        ctx.zipFile = new Zip();
+
+        if (ctx.instance.running()) {
+            ctx.instance.ui.log('Ghost is currently running. Backing up might take longer and slow down your site', 'yellow');
+            ctx.instance.loadRunningEnvironment();
```

Comment:
need to pass `true` to this function - otherwise if it's running in development the NODE_ENV variable won't get set, and the database backup will try to load the wrong config.